cocotb/share/lib/fli/FliImpl.cpp

```diff
@@ -434,6 +434,38 @@ void FliImpl::get_sim_precision(int32_t *precision)
     *precision = mti_GetResolutionLimit();
 }
 
+const char *FliImpl::get_simulator_product()
+{
+    if (m_product.empty()) {
+        char *c_info       = mti_GetProductVersion();      // Returned pointer must not be freed
+        std::string info   = c_info;
+        std::string search = " Version ";
+        std::size_t found  = info.find(search);
+
+        std::string product_str = c_info;
+        std::string version_str = c_info;
+
+        if (found != std::string::npos) {
+            product_str = info.substr(0, found);
+            version_str = info.substr(found+search.length());
+
+            LOG_DEBUG("Found Version string at %d", found);
+            LOG_DEBUG("   product: %s", product_str.c_str());
+            LOG_DEBUG("   version: %s", version_str.c_str());
+        }
+
+        m_product.assign(product_str.begin(), product_str.end());
+        m_version.assign(version_str.begin(), version_str.end());
```

Comment:
This code would be clearer without the intermediate `product_str` variable and using `operator=` instead of `.assign`, but I can see this is just moved code, so no need to tidy it in this PR.

Reply:
Tidied up anyways. Will test first.

Reply:
It builds and prints the correct string when running tests.